src/mongoc/mongoc-write-concern.c

```diff
@@ -98,7 +100,7 @@ bool
 mongoc_write_concern_get_fsync (const mongoc_write_concern_t *write_concern)
 {
    bson_return_val_if_fail(write_concern, false);
-   return write_concern->fsync_;
+   return (write_concern->fsync_ == true);
```

Comment:
They should probably be like "uint8_t fsync_ : 1;" anyway (just make sure unsigned). They probably weren't originally just to make it easier to assign since you can't take and address of a bitfield.